04/Task04.js

```diff
@@ -1,10 +1,51 @@
 import React from 'react';
+import Shop from '../src/components/Shop';
+import NotFound from '../src/components/NotFound';
+import products from '../src/products.json';
+import { Switch, Route, Redirect, useHistory } from 'react-router-dom';
 
 const Task04 = () => {
-    return (
-        <h1>Task04</h1>
-    );
-}
+  const history = useHistory();
 
-export default Task04;
+  const handleChange = (e) => {
+    history.push(e.target.value);
+
+    if (e.target.value === '/task04/price-desc') {
+      products.sort((a, b) => b.price - a.price);
+    } else if (e.target.value === '/task04/price-asc') {
+      products.sort((a, b) => a.price - b.price);
+    } else if (e.target.value === '/task04') {
+      products.sort((a, b) => a.id - b.id);
+    }
+  };
```

Comment:
Jakoś bardziej uniwersalnie to zrobić (czyli mnie ifów)? Pobierać dane z `e.target.value` i odwoływać się do funkcji w strukturze danych?
Podobnie jak tutaj: https://devmentor.pl/b/mniej-instrukcji-warunkowych#elastyczne-rozwi%C4%85zanie-%C5%82atwiejsze-rozszerzanie-kodu